### ISSUE TITLE
User Registration problems via JSON

### ISSUE THREAD

**Issue by boulderwebdev** [solid/node-solid-server#1023]
I want to be able to register users through the solid server api via the `/api/accounts/new` endpoint. Using python I could do the following:

```python
import requests

data = {
    'email': '...valid email',
    'username': '...valid username',
    'name': '...valid name',
    'password': '...valid password'
}
url = 'https://mysolidserver.com/api/accounts/new'
response = requests.post(url, data=data)
print(response.status_code) # which is 401
```
I am given a status code of 401 since I'm redirected to the `username.mysolidserver.com` page. This is awkward API UX since I'm expecting a confirmation response, probably in JSON with a status code in the 200's. Moreover, if I set `headers={'Content-Type': 'application/json'}` in the request, I am given an error claiming that the username is not valid.

Fortunately if the registration doesn't work, I'm given a status code of 400, so I can use this fact right now in my app.

**Comment by kjetilk** [comment]
Yeah, making the registration API better is something I would like to see, we just don't support it now. 

I think it is something that we should look into for "version.next", but we don't have a timeline for that yet.

**Comment by dmitrizagidulin** [comment]
@boulderwebdev The current user creation API only supports POSTs of type `application/x-www-form-urlencoded` (basically, a form submission), not JSON.
